File: website/src/scripts/JSONFilter.js
const convertToCsv = (data, fields) => {
  const headers = fields.join(',');
  const rows = data.map((row) =>
    fields.map((fieldName) => JSON.stringify(row[fieldName] || '')).join(',')
  );
  return [headers, ...rows].join('\n');
};

const processOddsData = (data) => {
  const rows = [];

  data.forEach((game) => {
    const sport = game.sport_title;
    const homeTeam = game.home_team;
    const awayTeam = game.away_team;
    const dateTimeOfGame = game.commence_time;

    game.bookmakers.forEach((bookmaker) => {
      const sportsbookName = bookmaker.title;

      bookmaker.markets.forEach((market) => {
        const betType = market.key;

        market.outcomes.forEach((outcome) => {
          if (betType === 'h2h') {
            rows.push({
              team1: outcome.name,
              team2: outcome.name === homeTeam ? awayTeam : homeTeam,
              betType: 'MoneyLine',
              betInfo: outcome.name,
              odds: outcome.price,
              dateOfGame: dateTimeOfGame,
              sportsbookName,
              sport,
            });
          } else if (betType === 'spreads') {
            rows.push({
              team1: outcome.name,
              team2: outcome.name === homeTeam ? awayTeam : homeTeam,
              betType: 'Spread',
              betInfo: `${outcome.point}`,
              odds: outcome.price,
              dateOfGame: dateTimeOfGame,
              sportsbookName,
              sport,
            });
          } else if (betType === 'totals') {
            rows.push({
              team1: homeTeam,
              team2: awayTeam,
              betType: 'Total',
              betInfo: `${outcome.name === 'Over' ? 'O' : 'U'}${outcome.point}`,
              odds: outcome.price,
              dateOfGame: dateTimeOfGame,
              sportsbookName,
              sport,
            });
          }
        });
      });
    });
  });

  const fields = ['team1', 'team2', 'betType', 'betInfo', 'odds', 'dateOfGame', 'sportsbookName', 'sport'];
  return convertToCsv(rows, fields); // Return the CSV string
};

export default processOddsData;

File: JSONFIlter.py
import pandas as pd
import json
from datetime import date

# Specify the file path
file_path = 'oddsapi_raw_data.json'

# Open and load the JSON file
with open(file_path, 'r') as file:
    data = json.load(file)  # Parses JSON into Python objects

# Get the current date
rows = []

for game in data:
    sport = game["sport_title"]
    home_team = game["home_team"]
    away_team = game["away_team"]
    datetime_of_game = game["commence_time"]
    
    for bookmaker in game["bookmakers"]:
        sportsbook_name = bookmaker["title"]
        
        for market in bookmaker["markets"]:
            bet_type = market["key"]
            
            for outcome in market["outcomes"]:
                if bet_type == "h2h":
                    # MoneyLine bets
                    bet_info = outcome["name"]
                    odds = outcome["price"]
                    rows.append([
                        outcome["name"],  # Team 1
                        away_team if outcome["name"] == home_team else home_team,  # Team 2
                        "MoneyLine", bet_info, odds, datetime_of_game, sportsbook_name, sport
                    ])
                elif bet_type == "spreads":
                    # Spread bets
                    bet_info = f"{outcome['point']}"
                    odds = outcome["price"]
                    rows.append([
                        outcome["name"],  # Team 1
                        away_team if outcome["name"] == home_team else home_team,  # Team 2
                        "Spread", bet_info, odds, datetime_of_game, sportsbook_name, sport
                    ])
                elif bet_type == "totals":
                    # Totals bets
                    bet_info = f"{'O' if outcome['name'] == 'Over' else 'U'}{outcome['point']}"
                    odds = outcome["price"]
                    rows.append([
                        home_team,  # Team 1
                        away_team,  # Team 2
                        "Total", bet_info, odds, datetime_of_game, sportsbook_name, sport
                    ])

# Create the DataFrame
df = pd.DataFrame(rows, columns=[
    "Team 1", "Team 2", "Bet Type", "Bet Info", "Odds", "Date of Game", "Sportsbook Name", "Sport"
])

df.to_csv('OddsAPI_combined_odds.csv', index=False)

